### PR TITLE
feat(togglegroup): add toggle group component 

### LINKED ISSUE #2625: Introduce Toogle Button Group
Introduce a new component called Toggle Button Group. This will be a groups of buttons that allow for selecting one items among a list of options (similar to a group of radio buttons). See design mockups below.

cc @mcoker @rachael-phillips @LHinson

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #2625 

new PR from new forked repo

old PR #3295

### CODE REVIEW

File: src/patternfly/components/ToggleGroup/examples/ToggleGroup.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+<br/>
+
+{{#> toggle-group}}
+  {{#> toggle-group-button toggle-group-button--modifier="pf-m-selected"}}
+    {{#> toggle-group-text}}
+      Option 1
+    {{/toggle-group-text}}
+  {{/toggle-group-button}}
+  {{> divider divider--type="div" divider--modifier="pf-m-vertical"}}
+  {{#> toggle-group-button toggle-group-button--modifier="pf-m-selected"}}
```
Comment (mattnolting) [inline src/patternfly/components/ToggleGroup/examples/ToggleGroup.md:41]:
```suggestion
  {{#> toggle-group-button}}
```
Looks like you have two set to selected

File: src/patternfly/components/ToggleGroup/toggle-group-icon.hbs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="pf-c-toggle-group__icon{{#if toggle-group-icon--modifier}} {{toggle-group-icon--modifier}}{{/if}}"
```
Comment (mcoker) [inline src/patternfly/components/ToggleGroup/toggle-group-icon.hbs:1]: can we make this a span?

File: src/patternfly/components/ToggleGroup/toggle-group-text.hbs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="pf-c-toggle-group__text{{#if toggle-group-text--modifier}} {{toggle-group-text--modifier}}{{/if}}"
```
Comment (mcoker) [inline src/patternfly/components/ToggleGroup/toggle-group-text.hbs:1]: same, can this be a span?

File: src/patternfly/components/ToggleGroup/toggle-group.scss [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    border-top-left-radius: var(--pf-c-toggle-group__button--first-child--BorderTopLeftRadius);
+    border-bottom-left-radius: var(--pf-c-toggle-group__button--first-child--BorderBottomLeftRadius);
+  }
+
+  &:last-child {
+    border-top-right-radius: var(--pf-c-toggle-group__button--last-child--BorderTopRightRadius);
+    border-bottom-right-radius: var(--pf-c-toggle-group__button--last-child--BorderBottomRightRadius);
+  }
+
+  &:hover {
```
Comment (mcoker) [inline src/patternfly/components/ToggleGroup/toggle-group.scss:51]: can we add a focus state?

File: src/patternfly/components/ToggleGroup/toggle-group.scss [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  display: flex;
+}
+
+.pf-c-toggle-group__button {
+  padding: var(--pf-c-toggle-group__button--PaddingTop) var(--pf-c-toggle-group__button--PaddingRight) var(--pf-c-toggle-group__button--PaddingBottom) var(--pf-c-toggle-group__button--PaddingLeft);
+  font-size: var(--pf-c-toggle-group__button--FontSize);
+  color: var(--pf-c-toggle-group__button--Color);
+  background-color: var(--pf-c-toggle-group__button--BackgroundColor);
+  border: 0;
```
Comment (mcoker) [inline src/patternfly/components/ToggleGroup/toggle-group.scss:35]: should this be an `inline-flex` item?

Comment (mcoker) [reply src/patternfly/components/ToggleGroup/toggle-group.scss:35]: Should we support icon positioning the same way we do with the button component? That would be another reason to use a button component here.

Comment (mattnolting) [inline src/patternfly/components/ToggleGroup/toggle-group.scss:35]:
```suggestion
  border: none;
```

File: src/patternfly/components/ToggleGroup/examples/ToggleGroup.md (hunk 1 above)
Comment (christiemolloy) [reply src/patternfly/components/ToggleGroup/examples/ToggleGroup.md:41]: this is to demonstrated two things can be clicked.

File: src/patternfly/components/ToggleGroup/toggle-group.scss (hunk 4 above)
Comment (christiemolloy) [reply src/patternfly/components/ToggleGroup/toggle-group.scss:51]: added the focus state and going to follow up with @mceledonia on monday

File: src/patternfly/components/ToggleGroup/toggle-group.scss (hunk 5 above)
Comment (christiemolloy) [reply src/patternfly/components/ToggleGroup/toggle-group.scss:35]: ^ will follow up on this

**Pull request merged**